packages/core/core/src/BundlerRunner.js

```diff
@@ -127,17 +127,19 @@ export default class BundlerRunner {
 
     // $FlowFixMe
     await dumpGraphToGraphViz(internalBundleGraph._graph, 'after_bundle');
-    try {
-      await bundler.optimize({
-        bundleGraph: mutableBundleGraph,
-        config: configResult?.config,
-        options: this.pluginOptions,
-        logger: new PluginLogger({origin: this.config.getBundlerName()}),
-      });
-    } catch (e) {
-      throw new ThrowableDiagnostic({
-        diagnostic: errorToDiagnostic(e, this.config.getBundlerName()),
-      });
+    if (this.pluginOptions.mode === 'production') {
```

Comment:
the signal check and `after_optimize` graphviz dump can be moved in here as well